recording/test/edge_timestamp_saver_test.exs

```diff
@@ -3,45 +3,69 @@ defmodule Membrane.RTC.Engine.Endpoint.Recording.EdgeTimestampSaverTest do
 
   import Membrane.Testing.Assertions
   import Membrane.ChildrenSpec
+  import MockTrack
 
   alias Membrane.Buffer
   alias Membrane.RTC.Engine.Endpoint.Recording.EdgeTimestampSaver
+  alias Membrane.RTCPEvent
   alias Membrane.Testing.{Pipeline, Sink, Source}
 
   tests_data = [
     %{
       title: "happy path",
       timestamps: Enum.to_list(1..200),
+      rtcp_events: 5,
       end_timestamps: [101, 200]
     },
     %{
       title: "edge timestamp not in order",
       timestamps: Enum.to_list(1..98) ++ [100, 99] ++ Enum.to_list(101..200),
+      rtcp_events: 5,
       end_timestamps: [101, 200]
     },
     %{
       title: "Long sequence not in order",
       timestamps: Enum.to_list(1..50) ++ Enum.to_list(100..10) ++ Enum.to_list(101..150),
+      rtcp_events: 5,
       end_timestamps: [101, 150]
     },
     %{
       title: "End is not in order",
       timestamps: Enum.to_list(1..50) ++ Enum.to_list(70..50),
+      rtcp_events: 5,
       end_timestamps: [70]
+    },
+    %{
+      title: "RTCP reports, before buffers",
+      timestamps: [],
+      rtcp_events: 5,
+      end_timestamps: []
     }
   ]
 
   for test_data <- tests_data do
     test test_data.title do
       pipeline = Pipeline.start_link_supervised!()
+      timestamps = unquote(test_data.timestamps)
+      events = unquote(test_data.rtcp_events)
 
       buffers =
-        for i <- unquote(test_data.timestamps) do
-          %Buffer{payload: <<>>, metadata: %{rtp: %{timestamp: i}}}
+        for i <- timestamps do
+          {:buffer, {:output, %Buffer{payload: <<>>, metadata: %{rtp: %{timestamp: i}}}}}
+        end
+
+      rtcp = create_rtcp_report(1, 1, 1)
+
+      events =
+        for _idx <- 0..events do
+          {:event, {:output, %RTCPEvent{rtcp: rtcp}}}
```

Comment:
This way, we don't know if the last `RTCPEvent` is saved or the last one, as we generate five of the same events.